File: Assets/Scripts/PatrolPoint.js
#pragma strict

public var patroller:GameObject;
private var activable:boolean = true;

function OnTriggerEnter(collider:Collider) {
	if(collider.gameObject.tag == "Enemy" && activable) {
		Debug.Log("Enemy Entered");
		patroller.SendMessage("SetNextPatrolPoint");
		activable = false;
	}
}

function OnTriggerExit(collider:Collider) {
	if(collider.gameObject.tag == "Enemy") {
		activable = true;
	}
}

File: Assets/Scripts/C#/PatrolPoint.cs

using UnityEngine;
using System.Collections;

public class PatrolPoint : MonoBehaviour {
	
	
	public GameObject patroller;
	private bool  activable = true;
	
	void  OnTriggerEnter ( Collider collider  ){
		if(collider.gameObject.tag == "Enemy" && activable) {
			patroller.SendMessage("SetNextPatrolPoint");
			activable = false;
		}
	}
	
	void  OnTriggerExit ( Collider collider  ){
		if(collider.gameObject.tag == "Enemy") {
			activable = true;
		}
	}
}